### DISCUSSION TITLE
train_text_to_image Poor training generation effect

### DISCUSSION THREAD

**Discussion by wzhiyuan2016** [huggingface/diffusers#10195] [General]
I use the train_text_to_image.py script to train UNet,Some of my relevant information is as follows:
1.My training script
![企业微信截图_17339701687867](https://github.com/user-attachments/assets/4ac73743-f510-4168-93e9-08831a7ef004)

2.My training data
![企业微信截图_17339702102638](https://github.com/user-attachments/assets/db3366fe-ffcd-4842-8c53-0b1626cbcb8d)
![企业微信截图_17339702641156](https://github.com/user-attachments/assets/e0ce1397-0860-40b5-a71a-40523981e1d6)

3.My training loss
![企业微信截图_17339703463573](https://github.com/user-attachments/assets/6a0979cf-322d-491f-ad24-c9f9c68ad5ab)
![企业微信截图_17339703722616](https://github.com/user-attachments/assets/ecfdf296-c435-4ccd-8e4a-e0ba00cb1fd8)

4.inference generating codes
![image](https://github.com/user-attachments/assets/2236a2b2-6c76-47fa-b949-a61944859954)

5.Generate results after training
prompt_list = ["lakes"]
![image](https://github.com/user-attachments/assets/007732cd-bfda-4ba3-992c-57e513dd2e7d)

6.Original["CompVis/stable-diffusion-v1-4"] generated results
prompt_list = ["lakes"]
![image](https://github.com/user-attachments/assets/2ddbfc2b-2c69-40d0-8694-bfb5114bab99)
My reasoning code is fine.

What is the reason why my generation effect has deteriorated? thank you!

**Comment by wzhiyuan2016** [comment]
@sayakpaul

**Reply by sayakpaul** [reply]
Stable Diffusion v1.4 doesn't take any crop coordinates. I am seeing a bunch of unsupported arguments in your training command. Could that be a problem?

**Reply by wzhiyuan2016** [reply]
@sayakpaul
Are you saying that I used these two parameters? 
--center_crop --random_flip

I saw it https://github.com/huggingface/diffusers/tree/main/examples/text_to_image
recommend using this parameter below. 
![企业微信截图_17339849694298](https://github.com/user-attachments/assets/6ea2b013-71df-463c-8266-d3804340791c)
So I used these two parameters

I will delete these two parameters and retrain them。

**Reply by sayakpaul** [reply]
Can you copy-paste the training command you're using?

**Reply by wzhiyuan2016** [reply]
@sayakpaul 
train_text_to_image.sh：
export MODEL_NAME="CompVis/stable-diffusion-v1-4"
export VAE_NAME="madebyollin/sdxl-vae-fp16-fix"
export TRAIN_DIR="./datas17/v1_tmp/"
export OUT_DIR="./train_snapshots/t2i_sd1.4_LHQ_unet_1210_tmp_del_center_crop_random_flip"
log=$1
LD_LIBRARY_PATH=./cudnn_8_7_0/cudnn-linux-x86_64-8.7.0.84_cuda11-archive/lib/ \
XFORMERS_MORE_DETAILS=1 \
accelerate launch --main_process_port 29501  --mixed_precision="fp16"  --multi_gpu --config_file "default_config_new2.yaml" train_text_to_image.py \
  --pretrained_model_name_or_path=$MODEL_NAME \
  --train_data_dir=$TRAIN_DIR \
  --caption_column="text" \
  --resolution=1024 \
  --train_batch_size=4 \
  --dataloader_num_workers=8 \
  --max_train_steps=20000 \
  --mixed_precision="fp16" \
  --validation_epochs=50 \
  --gradient_accumulation_steps=2 \
  --checkpointing_steps=5000 \
  --learning_rate=1e-05 \
  --lr_scheduler=constant \
  --lr_warmup_steps=10 \
  --seed=42 \
  --use_8bit_adam \
  --output_dir=$OUT_DIR \
  --gradient_checkpointing \
  --resume_from_checkpoint="latest" \
  --enable_xformers_memory_efficient_attention \
  --prediction_type="epsilon" 2>&1 | tee $log 
  
  default_config_new2.yaml:
  compute_environment: LOCAL_MACHINE
  debug: false
  distributed_type: MULTI_GPU
  downcast_bf16: 'no'
  gpu_ids: '4,5,6,7'
  machine_rank: 0
  main_training_function: main
  mixed_precision: fp16
  num_machines: 1
  num_processes: 4
  tpu_use_cluster: false
  tpu_use_sudo: false
  use_cpu: false

**Reply by sayakpaul** [reply]
Have you tried with a lower resolution like 512?

**Reply by wzhiyuan2016** [reply]
> Have you tried with a lower resolution like 512?

no, I'll try it out later.   Set the resolution to 512 for training  ,What are the points you are considering?
Training stability?-----> Lower resolution may make the training process more stable and reduce the risk of overfitting, as fewer details mean that the model needs to learn less complexity.
